pkg/apiserver/registry/generic/resource.go

```diff
@@ -39,6 +41,14 @@ func (info *ResourceInfo) WithGroupAndShortName(group string, shortName string)
 	}
 }
 
+func (info *ResourceInfo) KeyRootFunction(ctx context.Context) string {
```

Comment:
do these need to be meber functions of ResourceInfo?  can they be utility functions that are passed the resoure info?  then we would not need the apiserver dependency for all types

Reply:
they don't need to be, but it mostly depends on [this thread](https://github.com/grafana/grafana/pull/89635#discussion_r1651465474)

Reply:
moved to a partial application util function in https://github.com/grafana/grafana/pull/89635/commits/ec935986be833cf71819ae871d82f4d2e3c56a66